add opendirectoryd_version artifact to check for installation of Secu…

haircut opened the pull request:
…rity Update 2017-001 for High Sierra

SCCS project version checking as provided in https://support.apple.com/en-gb/HT208315

clburlison commented on artifacts/opendirectory_version.py:
```diff
@@ -0,0 +1,30 @@
+        stdout = None
+
+    if stdout:
+        result = stdout.splitlines()[-1].split(':')[-1]
```
To match other "version" facts in this repo I would change this to:
```python
result = stdout.splitlines()[-1].split('-')[-1]
```
```python
Old 
<result>opendirectoryd-483.20.7</result>
vs new
<result>483.20.7</result>
````

chilcote replied:
TIL about /usr/bin/what

As for the PR, I really don't care as long as it doesn't break the run :)

clburlison replied:
I must have skimmed past that. I personally would prefer your recommendation to make rename this artifact. Also didn’t even realize/ think to look at the --version output.

haircut commented:
I've pushed the two changes.

- Renamed the original subject artifact to `opendirectoryd_project_version_number.py` to more accurately reflect its purpose and output
- Added `opendirectoryd_build_number.py` to report on that separately

No conflicts during the run @chilcote ;)
